lib/Segments/DynamicSegments/Filters/WooCommerceNumberOfOrders.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace MailPoet\Segments\DynamicSegments\Filters;
+
+use MailPoet\Entities\DynamicSegmentFilterEntity;
+use MailPoet\Entities\SubscriberEntity;
+use MailPoetVendor\Carbon\Carbon;
+use MailPoetVendor\Doctrine\DBAL\Query\QueryBuilder;
+use MailPoetVendor\Doctrine\ORM\EntityManager;
+
+class WooCommerceNumberOfOrders implements Filter {
+  const ACTION_NUMBER_OF_ORDERS = 'numberOfOrders';
+
+  /** @var EntityManager */
+  private $entityManager;
+
+  public function __construct(EntityManager $entityManager) {
+    $this->entityManager = $entityManager;
+  }
+
+  public function apply(QueryBuilder $queryBuilder, DynamicSegmentFilterEntity $filter): QueryBuilder {
+    global $wpdb;
+    $subscribersTable = $this->entityManager->getClassMetadata(SubscriberEntity::class)->getTableName();
+    $filterData = $filter->getFilterData();
+    $type = $filterData->getParam('number_of_orders_type');
+    $count = $filterData->getParam('number_of_orders_count');
+    $days = $filterData->getParam('number_of_orders_days');
+
+    $date = Carbon::now()->subDays($days);
+
+    $queryBuilder->innerJoin(
+      $subscribersTable,
+      $wpdb->postmeta,
+      'postmeta',
+      "postmeta.meta_key = '_customer_user' AND $subscribersTable.wp_user_id=postmeta.meta_value"
+    )->leftJoin(
+      'postmeta',
+      $wpdb->posts,
+      'posts',
+      'posts.ID = postmeta.post_id AND posts.post_date >= :date AND postmeta.post_id NOT IN ( SELECT id FROM ' . $wpdb->posts . ' as p WHERE p.post_status IN ("wc-cancelled", "wc-failed"))'
+    )->setParameter(
+      'date', $date->toDateTimeString()
+    )->groupBy(
+      'inner_subscriber_id'
+    );
+
+    if ($type === '=') {
+      $queryBuilder->having("COUNT(posts.ID) = :count");
+    } elseif ($type === '>') {
+      $queryBuilder->having("COUNT(posts.ID) > :count");
+    } elseif ($type === '<') {
+      $queryBuilder->having("COUNT(posts.ID) < :count");
+    }
+
+    $queryBuilder->setParameter('count', $count);
```

Comment:
Be careful when using these parameters in dynamic segments. More dynamic segments will be combined into one big query. Users will be able to create a dynamic filter:
`I want all subscribers who made more than 10 orders in the last 100 days OR 5 orders in the last 10 days.`

That creates a big query and the parameter `count` would be twice in the query and that wouldn't work as intended.  We need to make the parameters unique. Look here how it is done: https://github.com/mailpoet/mailpoet/blob/5b45bdac1e0ba38e99d20aad7f6883c8863ba892/lib/Segments/DynamicSegments/Filters/EmailAction.php#L78

Reply:
Thanks for pointing this out. I just fixed it.